### PR TITLE
Migrate to pyproject to be a library

### LINKED ISSUE #19: Export as library
So it can be used instead of having to rely on submodule
~~Depends on #6~~

### PR DESCRIPTION
**Pull request by saschanaz**
Closes #19

### CODE REVIEW

**Comment by LuisMayo** [comment]
I've given this a test. It seems to work after I've changed an import as you can see. There are a couple of problems though.

1. It currently is conflicting main branch
2. It can be installed using the git URL, which is fine by me, but this can't be uploaded to pypi since it has an URL dependency (xx-ent-wiki-sm). To fix this we would have to either add functionality to download xx-ent-wiki-sm from within the code or replace spacy entirely.

So, to clarify things, for this PR to be merged it's fine if only point 1 is fixed, point 2 is not necessary for me. But if you want me to upload this to pypi it should be fixed too.

Thanks!

**Comment by saschanaz** [comment]
For now the git URL is also fine for my use. Perhaps the spacy thing can be done separately to limit the scope of this PR.

File: README.md [hunk 1]
```diff
@@ -44,12 +44,10 @@ Check the exmaple
 
 ### Using it as a library
  - Add it as a library with
-`git submodule add https://github.com/LuisMayo/objection_engine.git`
+`https://github.com/LuisMayo/objection_engine.git`
```
Comment (saschanaz) [inline README.md:47]: The depends on which project format the consumer uses, not sure the best way to describe that.

Comment (LuisMayo) [reply README.md:47]: Maybe we could give a couple of examples?

> If using pip
> `git+https://github.com/LuisMayo/objection_engine.git`
> If using X
> `EXAMPLE`

**Comment by LuisMayo** [comment]
This seems good to go now. Thanks a lot for the PR and for the patience you've shown during it.

**Pull request merged**